backend/frontend/views.py

```diff
@@ -294,6 +294,24 @@ def action_start_card(request, card_id):
         },
     )
 
+@user_passes_test(is_super)
+@csrf_exempt
+def link_card_details_page(request, card_id):
+    """The card is in the backlog and the user has chosen to start it"""
+    card = get_object_or_404(AgileCard, id=card_id)
+
+    recruit_project_reviews = get_list_or_404(RecruitProjectReview)
+    reviews = [{"timestamp":review.timestamp,"reviewer": get_object_or_404(User,id=review.reviewer_user_id).get_username(),"comments":review.comments,"status":review.status} for review in recruit_project_reviews if review.recruit_project_id == card.recruit_project_id]
```

Comment:
You are over-using the whole get...or_404 thing. 

We can just get a review user like this: `review.reviewer_user`. There is a constraint on the database so it's impossible for there to be a review without a reviewer user. Unless postgres or Django have massive bugs which is very unlikely.

Also, why are you doing this at all? You can just pass the reviews to the template. The context could be:
```
{
"project": project,
"reviews": project.project_reviews.order_by("- timestamp")
}
```

Then loop over the reviews inside the template in the normal way.

Reply:
Also, if a project has no reviews then we really should not raise a 404.  All projects start out with no reviews, and then people review them by accessing the project page.

If this page raises a 404 if there are no reviews on a project then nobody will be allowed to add the first review. Nobody will be allowed to see the link that was submitted even.

Please spend some time understanding what happens whien a 404 is raised on a page. It literally stops the whole page from working. It's a proper exception